Jellyfin.Api/Helpers/DynamicHlsHelper.cs

```diff
@@ -0,0 +1,549 @@
+            IMediaSourceManager mediaSourceManager,
+            IServerConfigurationManager serverConfigurationManager,
+            IMediaEncoder mediaEncoder,
+            IFileSystem fileSystem,
+            ISubtitleEncoder subtitleEncoder,
+            IConfiguration configuration,
+            IDeviceManager deviceManager,
+            TranscodingJobHelper transcodingJobHelper,
+            INetworkManager networkManager,
+            ILogger<DynamicHlsHelper> logger)
+        {
+            _libraryManager = libraryManager;
+            _userManager = userManager;
+            _dlnaManager = dlnaManager;
+            _authContext = authContext;
+            _mediaSourceManager = mediaSourceManager;
+            _serverConfigurationManager = serverConfigurationManager;
+            _mediaEncoder = mediaEncoder;
+            _fileSystem = fileSystem;
+            _subtitleEncoder = subtitleEncoder;
+            _configuration = configuration;
+            _deviceManager = deviceManager;
+            _transcodingJobHelper = transcodingJobHelper;
+            _networkManager = networkManager;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Get master hls playlist.
+        /// </summary>
+        /// <param name="controller">Requesting controller.</param>
+        /// <param name="transcodingJobType">Transcoding job type.</param>
+        /// <param name="streamingRequest">Streaming request dto.</param>
+        /// <param name="enableAdaptiveBitrateStreaming">Enable adaptive bitrate streaming.</param>
+        /// <returns>A <see cref="Task"/> containing the resulting <see cref="ActionResult"/>.</returns>
+        public async Task<ActionResult> GetMasterHlsPlaylist(
+            BaseJellyfinApiController controller,
+            TranscodingJobType transcodingJobType,
+            StreamingRequestDto streamingRequest,
+            bool enableAdaptiveBitrateStreaming)
+        {
+            var isHeadRequest = controller.Request.Method == WebRequestMethods.Http.Head;
+            var cancellationTokenSource = new CancellationTokenSource();
+            return await GetMasterPlaylistInternal(
+                controller,
+                streamingRequest,
+                isHeadRequest,
+                enableAdaptiveBitrateStreaming,
+                transcodingJobType,
+                cancellationTokenSource).ConfigureAwait(false);
+        }
+
+        private async Task<ActionResult> GetMasterPlaylistInternal(
+            BaseJellyfinApiController controller,
+            StreamingRequestDto streamingRequest,
+            bool isHeadRequest,
+            bool enableAdaptiveBitrateStreaming,
+            TranscodingJobType transcodingJobType,
+            CancellationTokenSource cancellationTokenSource)
+        {
+            using var state = await StreamingHelpers.GetStreamingState(
+                    streamingRequest,
+                    controller.Request,
```

Comment:
It seems this code is only ever accessing Request and Response. Both of which are available via `IHttpContextAccessor`. Please add it as a constructor argument and remove the dependency on `BaseJellyfinApiController`.

Since all three helpers are Scoped, `IHttpContextAccessor` will always contain the _current_ HttpContext.